### PR TITLE
 🐛 make sure we use the parsed argv

### PR DESCRIPTION
**Pull request by wei-lee**

### CODE REVIEW

File: lib/fhc.js [hunk 1]
```diff
@@ -152,7 +152,7 @@ fhc.applyCommandFunction = function (processArgv, cb) {
   // Only new-style commands can be validated
   if (cmdFn.demand) {
     try {
-      argUtils.validate(yargs, cmdFn);
+      argv = argUtils.validate(yargs, cmdFn);
```
Comment (wei-lee) [inline lib/fhc.js]: @camilamacedo86 As I mentioned in PR #307 , this is actually the real problem - we are not passing the new argv to the functions after this point.

File: lib/utils/args.js [hunk 2]
```diff
@@ -59,7 +59,7 @@ function _normaliseFlagsAndArs(argsIntoCommand, cmd){
     if (key.match(/[0-9]/)){
       var position = parseInt(key, 10),
         arg = argsIntoCommand[position];
-      var hasArgumentAtPosition = !isNaN(position) && typeof arg !== 'undefined' && !arg.match(/^--/);
+      var hasArgumentAtPosition = !isNaN(position) && typeof arg !== 'undefined' && !arg.match(/^--/) && arg.indexOf('=') === -1;
```
Comment (wei-lee) [inline lib/utils/args.js]: this is an extra check to make sure the value at this index is not like `-alias=key` or `--alias=key`

File: lib/fhc.js (hunk 1 above)
Comment (camilamacedo86) [inline lib/fhc.js]: @wei-lee Could you check the following commands with this PR? Are these commands working?

* bin/fhc.js app read --p=projectID --a=projectID
* bin/fhc.js app read -p=projectID -a=appID
* bin/fhc.js app read p=projectID a=appID
* bin/fhc.js app read projectID appID

Look that is the requirement into https://issues.jboss.org/browse/RHMAP-7260 which originated the Pr #307 . Please, if this alias are working now feel free to close the #307 in favor of this one.

Comment (wei-lee) [inline lib/fhc.js]: @camilamacedo86 only the third one is not supported. It is not the right way to specify aliases - it has to be prefixed with "-" or "--"

**Comment by camilamacedo86** [comment]
JIRA: https://issues.jboss.org/browse/RHMAP-8290

**Pull request merged**